### ISSUE TITLE
Release: 2.0.4

### ISSUE THREAD

**Issue by awoods187** [cockroachdb/cockroach#27395]
Candidate SHA: XXX EDIT ME XXX.
Deployment status: XXX EDIT ME XXX.

[Release process checklist](https://github.com/cockroachdb/cockroach/wiki/Release-Process)

Prep date: `XXX fill in date to roll out: usually Monday, a week before the release XXX`
- [ ] [Pick a sha](https://github.com/cockroachdb/cockroach/wiki/Release-Process#pick-sha), fill in `Candidate SHA` above.
- [ ] [Deploy to test clusters and start nightly suite](https://github.com/cockroachdb/cockroach/wiki/Release-Process#deploy-to-test-clusters)
- [ ] fill in `Deployment status` above with clusters and link to Nightly TeamCity Job
- [ ] [Verify node crash reports](https://github.com/cockroachdb/cockroach/wiki/Release-Process#verify-crash-reports)
- Keep an eye on clusters until release date. Do not proceed below until the release date.

Release date: `XXX fill in date: usually Monday XXX`
- [ ] [Check cluster status](https://github.com/cockroachdb/cockroach/wiki/Release-Process#check-cluster)
- [ ] [Tag release](https://github.com/cockroachdb/cockroach/wiki/Release-Process#tag-release)
- [ ] [Trigger release build](https://github.com/cockroachdb/cockroach/wiki/Release-Process#trigger-build)
- [ ] [Check binaries](https://github.com/cockroachdb/cockroach/wiki/Release-Process#check-binaries)

- For production or stable releases in the latest major release series
  - [ ] Update the [Homebrew Formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/cockroach.rb)
  - [ ] Update the image tag in our [Orchestrator configurations](https://github.com/cockroachdb/cockroach/wiki/Release-Process#orchestrator-configurations)
  - [ ] Update the version label in our [Red Hat Atomic Docker image](https://github.com/cockroachdb/cockroach/blob/master/build/deploy/rhel/Dockerfile). For major/minor releases, also update the documentation link in the [image's help file](https://github.com/cockroachdb/cockroach/blob/master/build/deploy/rhel/help.1)
  - [ ] Announce version to the [registration cluster](https://github.com/cockroachlabs/production#publishing-a-new-release-version)
  - [ ] Upgrade the registration cluster (reg runs latest stable version)

- [ ] Update docs
- [ ] External communications for release
- [ ] File issue for next release and assign to release coordinator

**Comment by bdarnell** [comment]
The candidate builds running on omega are identifying as `v2.0.3-48-gd7c9973-dirty`. I think the `-dirty` suffix means something has rotted in our build setup. @benesch can you take a look?

**Comment by benesch** [comment]
Indeed, the build that produced those binaries, https://teamcity.cockroachdb.com/viewLog.html?buildId=772993&buildTypeId=Cockroach_MergeToMaster&tab=buildLog, proudly declares that it's creating them with that tag:

```
 CC=/x-tools/x86_64-unknown-linux-gnu/bin/x86_64-unknown-linux-gnu-gcc CXX=/x-tools/x86_64-unknown-linux-gnu/bin/x86_64-unknown-linux-gnu-g++ go build -i -o ./cockroach-linux-2.6.32-gnu-amd64 -v  -installsuffix release-gnu -tags ' make x86_64_unknown_linux_gnu' -ldflags ' -s -w -extldflags "-static-libgcc -static-libstdc++" -X github.com/cockroachdb/cockroach/pkg/build.typ=release -X "github.com/cockroachdb/cockroach/pkg/build.tag=v2.0.3-48-gd7c9973-dirty" -X "github.com/cockroachdb/cockroach/pkg/build.utcTime=2018/07/11 15:55:33" -X "github.com/cockroachdb/cockroach/pkg/build.rev=d7c99735249c963177eadbc5fd57fc2094de6823" -X "github.com/cockroachdb/cockroach/pkg/build.cgoTargetTriple=x86_64-unknown-linux-gnu" -X "github.com/cockroachdb/cockroach/pkg/build.channel=official-binary" ' .
```

So this is less scary than the last time this -dirty nonsense happened. That time it was basically a race condition in Git that was to blame. I'm speculating here, but I suspect that the submodule churn resulted in an unclean state on this builder when this build executed. I've added a `git clean -ffdx` step to the beginning of the builder. Hopefully that takes care of it.

**Comment by bdarnell** [comment]
Sounds like we should do a test build before the final release, then. @benesch can you work with @awoods187 to make sure that happens?

**Comment by benesch** [comment]
Gah, looks like the test-publish build needs a tag name. @bdarnell are you OK with us tagging 2.0.4 today?

**Comment by bdarnell** [comment]
Yes, we should tag 2.0.4 today. For the test build, I wasn't thinking about tagging and using the teamcity test-publish build, just verifying that a build produced in the same way as the one we were testing doesn't get the `-dirty` tag.

**Comment by benesch** [comment]
Oh, sure, I can kick off another bleeding-edge build:
https://teamcity.cockroachdb.com/viewQueued.html?itemId=780683&tab=queuedBuildOverviewTab

The failure was nondeterministic, though, so even if this creates a binary
without -dirty it's not a guarantee that it won't happen on the next build.

**Comment by awoods187** [comment]
alright, then unless there are any objections, i'm going to start completing the checklist posted above @bdarnell @benesch
